src/api/aux.cpp

```diff
@@ -14,14 +14,19 @@
 
 
 /*** Function that reads JSON file with batch auction information and builds appropriate data structures with all the required information **/
-void parse_json_file(std::vector<Token> &tokens, std::vector<Order> &orders, std::vector<CP_AMM> &amms, int &_num_tokens, std::map<std::string, int> _idx_tokens)
+void parse_json_file(std::vector<Token> &tokens, std::vector<Order> &orders, std::vector<CP_AMM> &amms, int &_num_tokens, std::map<std::string, int> _idx_tokens, bool is_server_call, std::string server_input_string)
 {
     // TODO: Make input reading safe, i.e., handle all potential exceptions/errors
-    // TODO: Properly parse JSON and store sell/buy amounts using appropriate data types
 
     nlohmann::json json_file;
-    std::ifstream input_file("../../sample_instances/sample.json");
-    input_file >> json_file;
+
+    // Reading the input and creating JSON object, depending on whether solver or solver_server made the function call
+    if (is_server_call)
+        json_file = nlohmann::json::parse(server_input_string);
+    else {
+        std::ifstream input_file("./sample_instances/sample.json");
+        input_file >> json_file;
+    }
```

Comment:
I think it would be cleaner, if the parsing and the reading from input file are two different function. 
Then you can also avoid this bool flag as input parameter and always give the parsing function the string to parse

Reply:
You have a point here. I just wanted to not have any such code in the main file, but you are right that this flag variable is a dirty fix, so I will follow your suggestion.